packages/eslint-plugin/src/rules/no-useless-template-literals.ts

```diff
@@ -1,175 +1,20 @@
-
-                fixer.removeRange([
-                  node.expressions[0].range[1],
-                  nextQuasi.range[0] + 2,
-                ]),
-              ];
-            },
-          });
-
-          return;
-        }
-
-        const fixableExpressions = node.expressions.filter(
-          expression =>
-            isLiteral(expression) ||
-            isTemplateLiteral(expression) ||
-            isUndefinedIdentifier(expression) ||
-            isInfinityIdentifier(expression) ||
-            isNaNIdentifier(expression),
-        );
-
-        fixableExpressions.forEach(expression => {
-          context.report({
-            node: expression,
-            messageId: 'noUselessTemplateLiteral',
-            fix(fixer): TSESLint.RuleFix[] {
-              const index = node.expressions.indexOf(expression);
-              const prevQuasi = node.quasis[index];
-              const nextQuasi = node.quasis[index + 1];
-
-              // Remove the quasis' parts that are related to the current expression.
-              const fixes = [
-                fixer.removeRange([
-                  prevQuasi.range[1] - 2,
-                  expression.range[0],
-                ]),
-
-                fixer.removeRange([
-                  expression.range[1],
-                  nextQuasi.range[0] + 1,
-                ]),
-              ];
-
-              const stringValue = getStaticStringValue(expression);
-
-              if (stringValue != null) {
-                const escapedValue = stringValue.replace(/([`$\\])/g, '\\$1');
-
-                fixes.push(fixer.replaceText(expression, escapedValue));
-              } else if (isTemplateLiteral(expression)) {
-                // Note that some template literals get handled in the previous branch too.
-                // Remove the beginning and trailing backtick characters.
-                fixes.push(
-                  fixer.removeRange([
-                    expression.range[0],
-                    expression.range[0] + 1,
-                  ]),
-                  fixer.removeRange([
-                    expression.range[1] - 1,
-                    expression.range[1],
-                  ]),
-                );
-              }
-
-              return fixes;
-            },
-          });
-        });
-      },
-    };
-  },
-});
+import { createRule } from '../util';
+import type { NoUselessTemplateExpressionMessageId } from './no-useless-template-expression';
+import { makeNoUselessTemplateExpressionRuleObject } from './no-useless-template-expression';
+
+// This rule was renamed to `no-useless-template-expression`.
+// This module's purpose is just to import the code from the new implementation
+// and adjust its metadata to account for the renaming.
+// See https://github.com/typescript-eslint/typescript-eslint/issues/8544
+
+const ruleObject = makeNoUselessTemplateExpressionRuleObject();
+// @ts-expect-error: easier than figuring out how to make this mutable.
```

Comment:
[Docs] Nit: we know how, it's just annoying 😄 

```suggestion
// @ts-expect-error: easier than asserting this as mutable or deep object spreads.
```